sycl/doc/extensions/experimental/sycl_ext_oneapi_memory_export.asciidoc

```diff
@@ -0,0 +1,338 @@
+implementation supports.
+
+[frame="none",options="header"]
+|======================
+|Rev | Description
+|1   | Initial draft of the proposal
+|======================
+
+=== Querying device support
+
+We provide the following device aspect to query for support of exporting memory.
+
+[frame="none",options="header"]
+|======================
+|Device descriptor |Description
+|`aspect::ext_oneapi_memory_export_linear` | Indicates if the device supports
+the allocation of exportable linear memory and exporting that memory to an
+interoperable handle.
+|======================
+
+=== External Memory Resource Handle Types [[external_mem_res_handles]]
+
+This extension supports two types of external memory resource handles. These
+types are used to import the memory allocated by this extension into external
+APIs.
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+
+// External memory resource handle types.
+enum class export_external_mem_handle_type {
+  opaque_fd = 0,
+  win32_nt = 1,
+};
+
+template <export_external_mem_handle_type ExportMemHandleType>
+struct exported_mem;
+
+template <>
+struct exported_mem<export_external_mem_handle_type::opaque_fd> {
+  using type = int;
+};
+
+template <>
+struct exported_mem<export_external_mem_handle_type::win32_nt> {
+  using type = void *;
+};
+
+template <export_external_mem_handle_type ExportMemHandleType>
+using exported_mem_t = typename exported_mem<ExportMemHandleType>::type;
+
+}
+```
+
+The `export_external_mem_handle_type` enum class defines the types of external
+memory resource handles that can be exported by this extension. The `opaque_fd`
+and `win32_nt` values are used during allocation of exportable memory to
+indicate the type of handle that will later be returned by the
+`export_device_mem_handle` function.
+
+The `opaque_fd` handle type corresponds to a POSIX file descriptor, which is
+represented by an `int`.
+
+The `win32_nt` handle type corresponds to a Windows NT handle, which is
+represented by a `void *`.
+
+The `exported_mem` template struct is specialized for each of the
+`export_external_mem_handle_type` values. Each specialization defines a `type`
+alias that represents the type of the exported memory handle for that specific
+handle.
+
+The `exported_mem_t` type alias allows the user to easily obtain the type of
+the exported memory handle based on the `export_external_mem_handle_type`
+template parameter.
+
+=== API of the extension
+
+```c++
+
+namespace sycl::ext::oneapi::experimental {
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    export_external_mem_handle_type exportMemHandleType,
+    const sycl::device &syclDevice, const sycl::context &syclContext,
+    const property_list& propList = {});
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    export_external_mem_handle_type exportMemHandleType,
+    const sycl::queue &syclQueue,
+    const property_list& propList = {});
+}
+```
```

Comment:
Why are we adding these new APIs when the existing USM functions accept a property list?

We could use something like:
```c++
auto properties = { syclx::property::memory::exportable(syclx::mem_handle::opaqe_fd) };
auto ptr = aligned_alloc_device(alignment, size, q, properties);
```

I agree it's more verbose, but that's because I'm deliberately trying to follow SYCL 2020 style. There are probably ways to make it shorter.  The advantage of this approach is that you'd be able to use all the other `malloc_device` overloads to get exportable memory as well.

Reply:
Addressed here: https://github.com/intel/llvm/pull/19018#discussion_r2182052526